test/integration/pkcs11-tool.sh

```diff
@@ -73,6 +73,37 @@ echo "Deleting pubkey"
 pkcs11_tool --slot=1 --pin=myuserpin --login --delete-object --type=pubkey --label=myecckey
 echo "Pubkey deleted"
 
+# test EC derive
+echo "Generating a shared secret"
+echo "Create EC keys"
+pkcs11_tool --slot=1 --keypairgen --login --pin myuserpin \
+            --key-type EC:prime256v1 --id 01 --label key1
+
+pkcs11_tool --slot=1 --keypairgen --login --pin myuserpin \
+            --key-type EC:prime256v1 --id 02 --label key2
+
+echo "Read public components"
+pkcs11_tool --read-object --login --pin myuserpin --type pubkey -id 01 \
+            -o /tmp/key1_pub.der
+
+pkcs11_tool --read-object --login --pin myuserpin --type pubkey --id 02 \
+            -o /tmp/key2_pub.der
+
+echo "Derive secrets"
+pkcs11_tool  --derive -m ECDH1-DERIVE --id 01 --label key1 \
+             --login --pin myuserpin \
+             --input-file /tmp/key2_pub.der \
+             --output-file /tmp/shared_secret_1
+
+pkcs11_tool  --derive -m ECDH1-DERIVE --id 02 --label key2 \
+             --login --pin myuserpin \
+             --input-file /tmp/key1_pub.der \
+             --output-file /tmp/shared_secret_2
```

Comment:
We have all sorts of issues with pkcs11-tool, it's likely failing on certain versions and distros, we have skips in our tests because we never figured out all the crashes. Perhaps skip this on failing versions?

Reply:
ah ok.
yeah I was wondering...pkcs11-tool is kind of a Russian roulette. Ok I'll test if it can pull the public component from the key and skip if it fails.